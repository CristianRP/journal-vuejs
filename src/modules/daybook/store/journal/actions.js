import journalApi from '@/api/journalApi'
// export const myAction = async({ commit }) => {


// }

export const loadEntries = async({ commit }) => {
  const { data } = await journalApi.get('/entries.json')

  if ( !data ) {
    commit('setEntries', [])
  }

  const entries = []
  for (let id of Object.keys(data)) {
    entries.push({
      id,
      ...data[id]
    })
  }

  commit('setEntries', entries)
}

export const updateEntry = async({ commit }, entry) => { // entry should be a param

  // exctract only what we need // -id
  const { date, text } = entry
  const dataToSave = { date, text}

  // await journalApi.put (path, .json, dataToSave)
  await journalApi.put(`/entries/${entry.id}.json`, dataToSave)

  // Commit a mutation
  commit('updateEntry', entry)
}

export const createEntry = async({ commit }, entry) => {
  const { date, text } = entry
  const dataToSave = { date, text }

  const { data } = await journalApi.post(`/entries.json`, dataToSave)

  dataToSave.id = data.name

  commit('addEntry', dataToSave)

  return data.name
}

export const deleteEntry = async({ commit }, id) => {

  await journalApi.delete(`/entries/${id}.json`)

  commit('deleteEntry', id)

  return id
}
