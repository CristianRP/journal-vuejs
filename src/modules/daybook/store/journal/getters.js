// export const myGetter = (state) => {
  // return state
// }

export const getEntrieByTerm = (/* state */) => {
  // return state
}

export const getEntryById = (/* state */) => {
  // return state
}


