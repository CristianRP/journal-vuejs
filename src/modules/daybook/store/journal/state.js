export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: 'Labore dolor laborum aliqua est excepteur labore Lorem. Laborum sint elit sit culpa labore. Eiusmod id et amet elit exercitation. Eu amet deserunt commodo voluptate et. Consequat cupidatat aliqua id eu tempor mollit laborum sunt cupidatat ipsum Lorem. Reprehenderit labore do amet proident adipisicing nisi exercitation. Consectetur do magna nulla ut eiusmod do nulla veniam officia laborum et cillum tempor.',
      picture: null, // https:
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: 'Elit reprehenderit sit voluptate velit proident non labore pariatur dolore. Nisi aute amet adipisicing sint. Magna commodo minim ea reprehenderit minim cupidatat. Quis dolore incididunt reprehenderit enim.',
      picture: null, // https:
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: 'Veniam nisi ipsum laborum cupidatat eu ea. Elit et sit aute aute laborum incididunt incididunt fugiat ullamco labore eu incididunt. Lorem aliqua ad id sunt cillum excepteur consequat. Qui elit qui veniam elit ex.',
      picture: null, // https:
    }
  ]
})
