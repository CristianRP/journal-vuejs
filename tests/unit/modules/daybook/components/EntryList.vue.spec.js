import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'

import EntryList from '@/modules/daybook/components/EntryList'

import journal from '@/modules/daybook/store/journal'
import { getEntriesByTerm } from '@/modules/daybook/store/journal/getters'
import { journalState } from '../../../mock-data/test-journal-state'

const createVuexStore = ( initialStore ) => createStore({
  modules: {
    journal: {
      ...journal,
      state: { ...initialStore }
    }
  }
})

describe('Tests on EntryList', () => {

  let wrapper

  const store = createVuexStore( journalState )
  const mockRouter = {
    push: jest.fn()
  }

  beforeEach(() => {
    jest.clearAllMocks()

    wrapper = shallowMount( EntryList, {
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [ store ]
      }
    })
  })

  test('should call getEntriesByTerm and show 2 entries', () => {
    expect( wrapper.findAll('entry-item-stub').length ).toBe(2)
    expect( wrapper.html() ).toMatchSnapshot()
  })

  test('should call getEntriesByTerm and filter entries', async() => {
    const input = wrapper.find('input')
    await input.setValue('Test')

    expect( wrapper.findAll('entry-item-stub').length ).toBe(1)
  })

  test('new button should redirect to /new', () => {
    wrapper.find('button').trigger('click')

    expect( mockRouter.push ).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: 'new'
      }
    })
  })
})
