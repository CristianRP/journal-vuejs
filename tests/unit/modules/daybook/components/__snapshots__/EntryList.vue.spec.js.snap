// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on EntryList should call getEntriesByTerm and show 2 entries 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Search Entry"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> New Entry </button></div>
  <div class="entry-scrollarea">
    <entry-item-stub entry="[object Object]"></entry-item-stub>
    <entry-item-stub entry="[object Object]"></entry-item-stub>
  </div>
</div>
`;
